File: backend/src/TarlaAsistani.Domain/Entities/CaseMessage.cs
using TarlaAsistani.Domain.Enums;

namespace TarlaAsistani.Domain.Entities;

public class CaseMessage
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid CaseId { get; set; }

    public Guid SenderId { get; set; }

    public CaseMessageType MessageType { get; set; } = CaseMessageType.Comment;

    public string Body { get; set; } = null!;

    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

    // Navigation
    public SupportCase Case { get; set; } = null!;

    public User Sender { get; set; } = null!;

    public ICollection<CaseMessageMedia> MediaLinks { get; set; } = new List<CaseMessageMedia>();
}

File: mobile/lib/features/cases/domain/models/case_message.dart
enum CaseMessageType {
  comment,
  additionalInfoRequest,
  expertResponse;

  static CaseMessageType fromString(String? val) => switch (val?.toLowerCase()) {
    'additionalinforequest' => CaseMessageType.additionalInfoRequest,
    'expertresponse' => CaseMessageType.expertResponse,
    _ => CaseMessageType.comment,
  };
}

class CaseMessage {
  const CaseMessage({
    required this.id,
    required this.caseId,
    required this.senderId,
    required this.senderName,
    required this.messageType,
    required this.body,
    this.mediaUrls = const [],
    required this.createdAt,
    this.isCurrentUser = false,
  });

  final String id;
  final String caseId;
  final String senderId;
  final String senderName;
  final CaseMessageType messageType;
  final String body;
  final List<String> mediaUrls;
  final DateTime createdAt;
  final bool isCurrentUser;

  bool get isFromExpert =>
      messageType == CaseMessageType.expertResponse ||
      messageType == CaseMessageType.additionalInfoRequest;
}
